Add helm-pg-redis: HA-ready PostgreSQL/Redis Helm presets for GitGuardian (#33)
GitGuardian/ggtools: helm-pg-redis/README.md

### Helm: PostgreSQL and Redis for GitGuardian

This folder provides ready-to-use Helm configurations to deploy PostgreSQL and Redis in your Kubernetes cluster for use with the GitGuardian application.

- PostgreSQL topologies:
  - **HA (replication)**: primary with read replica(s); recommended for production for resilience and read scaling.
  - **Standalone**: single-primary setup; suitable for PoC or testing only.
- Redis topology:
  - **Standalone** with persistence (aligned with the scaling guide). If you require Redis replication/sentinel, extend these values as needed.
- All presets are available in three sizes aligned with the GitGuardian scaling guide: **small**, **medium**, **large**.

### Recommendation for cloud providers

If you deploy GitGuardian on a public cloud, prefer the provider's managed services for PostgreSQL and Redis instead of running them in-cluster:

- AWS: Amazon RDS/Aurora (PostgreSQL), Amazon ElastiCache/MemoryDB (Redis)

> - GCP: Cloud SQL for PostgreSQL, Memorystore for Redis

Author: ```suggestion
- GCP: [Cloud SQL for PostgreSQL](https://docs.gitguardian.com/self-hosting/installation/databases/postgres-cloudsql), [Memorystore for Redis](https://docs.gitguardian.com/self-hosting/installation/databases/redis-memorystore)
```

> - Azure: Azure Database for PostgreSQL, Azure Cache for Redis

Author: ```suggestion
- Azure: [Azure Database for PostgreSQL](https://docs.gitguardian.com/self-hosting/installation/databases/postgres-azure), [Azure Cache for Redis](https://docs.gitguardian.com/self-hosting/installation/databases/redis-azure-cache)
```

⚠️ Deploying PostgreSQL/Redis using these Bitnami Helm examples on Red Hat OpenShift is not supported.

Presets are mapped as follows:

- PostgreSQL
  - Small: Primary 4 vCPU / 8 GiB / 200 Gi; Read replica 2 vCPU / 4 GiB / 200 Gi
  - Medium: Primary 8 vCPU / 32 GiB / 250 Gi; Read replica 4 vCPU / 16 GiB / 250 Gi
  - Large: Primary 16 vCPU / 64 GiB / 300 Gi; Read replica 8 vCPU / 32 GiB / 300 Gi
- Redis (standalone)
  - Small: 2 vCPU / 2 GiB / 20 Gi
  - Medium: 4 vCPU / 8 GiB / 40 Gi
  - Large: 8 vCPU / 16 GiB / 100 Gi

For broader infrastructure guidance and context on these sizes, see the GitGuardian Scaling guide: [Scaling](https://docs.gitguardian.com/self-hosting/management/infrastructure-management/scaling).

You can customize storage classes, resource requests/limits, and replica counts by editing the values files or using `--set` overrides.

### Prerequisites

- Helm 3.x and `kubectl` configured against your target cluster
- A default `StorageClass` or an explicit one you will set in values files
- Cluster capacity matching the selected preset(s)

> 

Author: ```suggestion
- A Kubernetes namespace where PostgreSQL, Redis, and the GitGuardian application will be installed.
  - Example: `kubectl create ns gitguardian`

```

### Quick start

> Use the commands below to install Bitnami charts with the preset values files. Create the target namespace if missing.

Author: ```suggestion
Use the commands below to install Bitnami charts with the preset values files. Replace `<namespace>` accordingly.
```

1) Add Bitnami repo (if not already added) and create a namespace:

```bash
helm repo add bitnami https://charts.bitnami.com/bitnami && helm repo update
kubectl get ns gg-datastores >/dev/null 2>&1 || kubectl create ns gg-datastores
```

2) Optional: customize a values file

Edit the YAML presets under `values/postgres/` (and `values/redis/` later) to fit your environment. Typical edits include `auth.username`, `auth.database`, `persistence.storageClass`, `persistence.size`, and `resources`.

3) Install PostgreSQL (choose topology and size):

> 

Reviewer: ```suggestion

```bash
helm template --namespace <namespace> -f local-values.yaml -s templates/image-pull-secrets.yaml oci://registry.replicated.com/gitguardian/gitguardian > local_secrets.yaml
kubectl --namespace <namespace> apply -f ./local_secrets.yaml
```

```

```bash
# Standalone (PoC/testing) - small preset
helm upgrade --install pg bitnami/postgresql \
  -n gg-datastores \
  -f helm-pg-redis/values/postgres/standalone-small.yaml \
  --wait

# HA (recommended for production) - medium preset
helm upgrade --install pg bitnami/postgresql \

>   -n gg-datastores \

Author: ```suggestion
  -n <namespace> \
```

  -f helm-pg-redis/values/postgres/ha-medium.yaml \
  --wait
```

4) Install Redis (choose size):

```bash
# Standalone - small preset
helm upgrade --install redis bitnami/redis \
  -n gg-datastores \
  -f helm-pg-redis/values/redis/standalone-small.yaml \
  --wait

# Standalone - large preset
helm upgrade --install redis bitnami/redis \

>   -n gg-datastores \

Author: ```suggestion
  -n <namespace> \
```

  -f helm-pg-redis/values/redis/standalone-large.yaml \
  --wait
```

5) Retrieve credentials and assemble connection strings:

```bash

> NAMESPACE=gg-datastores

Author: ```suggestion
NAMESPACE=<namespace>
```
